packages/core/src/editor/BlockNoteEditor.ts

```diff
@@ -1008,6 +1090,11 @@ export class BlockNoteEditor<
   public onChange(
     callback: (editor: BlockNoteEditor<BSchema, ISchema, SSchema>) => void
   ) {
+    if (this.headless) {
+      // TODO: should actually be possible in headless mode as well
```

Comment:
Is there a use case for this though? Since there's not really any editing happening in headless mode.